### PR TITLE
upstream/downstream requests should get 404 (instead of 401) 

### PR DESCRIPTION
**Pull request by jitsni**
upstream/downstream requests should get 404 (instead of 401) for unbound wse sessions of a protected service

### CODE REVIEW

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.origin.server.response.streaming.rpt [hunk 1]
```diff
@@ -5,7 +5,7 @@
 connect tcp://localhost:8110
 connected
 
-write "GET /shield/;e/db/AkDKBgEzhWxUvxWu8Jsqnf6qCeAm8TNt HTTP/1.1\r\n"
+write "GET /shield HTTP/1.1\r\n"
```
Comment (cmebarrow) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.origin.server.response.streaming.rpt]: Why is this changed?

Comment (jitsni) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.origin.server.response.streaming.rpt]: I will rework on this and allow proxying for wse requests.

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpPathMatchingFilter.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        String nextProtocol = bindAddress.getOption(ResourceAddress.NEXT_PROTOCOL);
+        if (nextProtocol == null || !nextProtocol.equals("httpxe/1.1")) {
+            super.httpRequestReceived(nextFilter, session, httpRequest);
+            return;
+        }
+
+        String bindPath = bindAddress.getResource().getPath();
+        String requestPath = httpRequest.getRequestURI().getPath();
+        // Fix requestPath to overcome javascript client bug (which sends URI with & )
+        // for e.g. /echo/;e/utm/waGcY3k5CDLMxy0mNvijdpXRtHxFjt5Z&.krn=0.7393842079985082)
```
Comment (cmebarrow) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpPathMatchingFilter.java]: Please include a issue link (https://github.com/kaazing-private/gateway.client.javascript/issues/4) so we can remove this code when the issue is fixed and we no longer need support the current client version.

File: transport/wseb/src/test/scripts/org/kaazing/gateway/transport/wseb/send404OnInvalidDownstreamToProtectedService.rpt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2007-2015, Kaazing Corporation. All rights reserved.
+#
+
+connect http://localhost:8001/echoAuth/;e/db
```
Comment (cmebarrow) [inline transport/wseb/src/test/scripts/org/kaazing/gateway/transport/wseb/send404OnInvalidDownstreamToProtectedService.rpt]: Do we need extra test case where they put in a sessionid which is no longer valid (e.g. /echo/;e/utm/waGcY3k5CDLMxy0mNvijdpXRtHxFjt5Z or /echo/;e/utm/waGcY3k5CDLMxy0mNvijdpXRtHxFjt5Z&.krn=0.7393842079985082)?

**Comment by cmebarrow** [comment]
@jitsni Done reviewing, looks good, just couple of minor questions/points

**Comment by cmebarrow** [comment]
@jitsni Please respond to my remaining two comments then I can merge
